### PR TITLE
MDEXP-627 - /data-export/export: Authority Strategy

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MDEXP-627](https://issues.folio.org/browse/MDEXP-627) - /data-export/export: Authority Strategy

## Purpose
Existing AuthorityStrategy https://github.com/folio-org/mod-data-export/blob/master/src/main/java/org/folio/service/manager/export/strategy/AuthorityExportStrategyImpl.java should be implemented and integrated in the business flow with using interface https://github.com/folio-org/mod-data-export/blob/mdexp-spring/src/main/java/org/folio/dataexp/service/export/strategies/ExportStrategy.java.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] Check logging
  - [ ] There are no major code smells or security issues

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/dataexp/service/ConsortiaService.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class ConsortiaService {
+
+  @Value("${myuniversity}")
+  private String tenantId;
```
Comment (alekGbuz) [inline src/main/java/org/folio/dataexp/service/ConsortiaService.java:17]: tenantId will be not found in multiple tenant environment, to get tenant information please use FolioExecutionContext bean.

File: src/main/java/org/folio/dataexp/service/export/strategies/AuthorityExportStrategy.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@Component
+@RequiredArgsConstructor
+public class AuthorityExportStrategy extends AbstractExportStrategy {
+
+  private final RuleFactory ruleFactory;
+  private final ConsortiaService consortiaService;
+  private final MarcAuthorityRecordRepository marcAuthorityRecordRepository;
+
+  @Value("${myuniversity}")
+  private String tenantId;
```
Comment (alekGbuz) [inline src/main/java/org/folio/dataexp/service/export/strategies/AuthorityExportStrategy.java:30]: tenantId will be not found in multiple tenant environment, to get tenant information please use FolioExecutionContext bean.

Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/service/export/strategies/AuthorityExportStrategy.java]: Good catch, thank you.

File: src/main/java/org/folio/dataexp/service/ConsortiaService.java (hunk 1 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/service/ConsortiaService.java]: Good catch, thank you.

**Review by alekGbuz — APPROVED** [review]

**Review by khandramai — APPROVED** [review]

**Pull request merged**